src/simsopt/geo/biotsavart.py

```diff
@@ -54,58 +82,66 @@ def compute_A(self, points, compute_derivatives=0):
 
         self._A = sum(self.coil_currents[i] * self._dA_by_dcoilcurrents[i] for i in range(len(self.coil_currents)))
         if compute_derivatives >= 1:
-            self._dA_by_dX = sum(self.coil_currents[i] * self._d2A_by_dXdcoilcurrents[i] for i in range(len(self.coil_currents)))
+            self._dA = sum(self.coil_currents[i] * self._d2A_by_dXdcoilcurrents[i] for i in range(len(self.coil_currents)))
         if compute_derivatives >= 2:
-            self._d2A_by_dXdX = sum(self.coil_currents[i] * self._d3A_by_dXdXdcoilcurrents[i] for i in range(len(self.coil_currents)))
+            self._ddA = sum(self.coil_currents[i] * self._d3A_by_dXdXdcoilcurrents[i] for i in range(len(self.coil_currents)))
         return self
 
-    def compute(self, points, compute_derivatives=0):
-        assert compute_derivatives <= 2
-
-        self._dB_by_dcoilcurrents = [np.zeros((len(points), 3)) for coil in self.coils]
+    def _A_impl(self, A):
+        self.compute_A(compute_derivatives=0)
+        A[:] = self._A
```

Comment:
Passing a variable and filing it is more of C/C++ style. I am guessing we are doing it this way for efficiency's sake. Is there a python approach that is equally efficient? Also, slicing can create a copy. Please make sure there are no gotcha's here.

Reply:
Yeah, we're doing this so that the memory management can be done manually on the C++ side and so that we don't have to malloc a new array every time.

According to https://stackoverflow.com/questions/47181092/numpy-views-vs-copy-by-slicing, slicing only creates a copy when non-contiguous data is accessed, so I think we're good here.